zerver/actions/custom_profile_fields.py

```diff
@@ -187,9 +193,13 @@ def do_update_user_custom_profile_data_if_changed(
             )
 
 
-def check_remove_custom_profile_field_value(user_profile: UserProfile, field_id: int) -> None:
+def check_remove_custom_profile_field_value(
+    user_profile: UserProfile, field_id: int, acting_user: UserProfile
+) -> None:
     try:
         custom_profile_field = CustomProfileField.objects.get(realm=user_profile.realm, id=field_id)
+        if not acting_user.is_realm_admin and not custom_profile_field.editable_by_user:
+            raise JsonableError(_("User does not have permission to update the field."))
         field_value = CustomProfileFieldValue.objects.get(
```

Comment:
Maybe put a blank line after the check, to indicate we're moving onto a next thought.

Reply:
Added blank line as suggested.